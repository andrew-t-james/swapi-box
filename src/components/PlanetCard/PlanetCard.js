import React from 'react';
import Button from '../Button/Button';
import PropTypes from 'prop-types';

const PlanetCard = ({ climate, name, population, favorite, residents, terrain, updateFavorites }) => {
  const planet = {
    name,
    favorite
  };

  return (
    <div>
      <h2>{name}</h2>
      <p>{climate}</p>
      <p>{population}</p>
      <p>{residents}</p>
      <p>{terrain}</p>
      <span className="card__favorite" onClick={() => updateFavorites(planet)}>
        <Button name="⭐" />
      </span>
    </div>
  );
};

PlanetCard.propTypes = {
  climate: PropTypes.string,
  name: PropTypes.string,
  population: PropTypes.string,
  residents: PropTypes.array,
  terrain: PropTypes.string,
  favorite: PropTypes.bool,
  updateFavorites: PropTypes.func
};

export default PlanetCard;
