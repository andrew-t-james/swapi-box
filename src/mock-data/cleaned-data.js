export const cleanedPeople = [
  {
    name: 'Luke Skywalker',
    homeworld: 'Tatooine',
    population: '200000',
    species: 'Human',
    favorite: false
  }
];

export const peopleDataWithName = [
  {
    name: 'Luke Skywalker',
    homeworld: undefined,
    population: undefined,
    species: undefined,
    favorite: false
  }
];

export const mockCleanedMovie = {
  crawl: "It is a period of civil warRebel spaceships strikingfrom a hidden base have wontheir first victory againstthe evil Galactic EmpireDuring the battle Rebelspies managed to steal secretplans to the Empiresultimate weapon the DEATHSTAR an armored spacestation with enough powerto destroy an entire planetPursued by the Empiressinister agents PrincessLeia races home aboard herstarship custodian of thestolen plans that can save herpeople and restorefreedom to the galaxy",
  episode: 4,
  releaseDate: "1977-05-25",
  name: "A New Hope"
};

export const mockCleanedPlanet = [
  {
    name: 'Alderaan',
    population: '2000000000',
    climate: 'temperate',
    terrain: 'grasslands, mountains',
    residents: [undefined, undefined, undefined],
    favorite: false
  }
];

export const mockCleanedVehicle = [
  {
    name: 'Sand Crawler',
    model: 'Digger Crawler',
    vehicleClass: 'wheeled',
    passengers: '30',
    favorite: false
  }
];

export const mockCleanedFavorite = [
  {
    name: 'Sand Crawler'
  }
];