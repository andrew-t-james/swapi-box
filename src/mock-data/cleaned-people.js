export default [
  {
    name: 'Luke Skywalker',
    homeworld: 'Tatooine',
    population: '200000',
    species: 'Human'
  },
  {
    name: 'C-3PO',
    homeworld: 'Tatooine',
    population: '200000',
    species: 'Droid'
  },
  {
    name: 'R2-D2',
    homeworld: 'Naboo',
    population: '4500000000',
    species: 'Droid'
  },
  {
    name: 'Darth Vader',
    homeworld: 'Tatooine',
    population: '200000',
    species: 'Human'
  },
  {
    name: 'Leia Organa',
    homeworld: 'Alderaan',
    population: '2000000000',
    species: 'Human'
  },
  {
    name: 'Owen Lars',
    homeworld: 'Tatooine',
    population: '200000',
    species: 'Human'
  },
  {
    name: 'Beru Whitesun lars',
    homeworld: 'Tatooine',
    population: '200000',
    species: 'Human'
  },
  {
    name: 'R5-D4',
    homeworld: 'Tatooine',
    population: '200000',
    species: 'Droid'
  },
  {
    name: 'Biggs Darklighter',
    homeworld: 'Tatooine',
    population: '200000',
    species: 'Human'
  },
  {
    name: 'Obi-Wan Kenobi',
    homeworld: 'Stewjon',
    population: 'unknown',
    species: 'Human'
  }
];