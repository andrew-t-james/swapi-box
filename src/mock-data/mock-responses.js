export const mockCleanHomeWorldResponse = [
  {
    birth_year: "19BBY",
    created: "2014-12-09T13:50:51.644000Z",
    edited: "2014-12-20T21:17:56.891000Z",
    eye_color: "blue",
    films: [
      "https://swapi.co/api/films/2/",
      "https://swapi.co/api/films/6/",
      "https://swapi.co/api/films/3/",
      "https://swapi.co/api/films/1/",
      "https://swapi.co/api/films/7/"
    ],
    gender: "male",
    hair_color: "blond",
    height: "172",
    homeworld: "Tatooine",
    mass: "77",
    name: "Luke Skywalker",
    population: "200000",
    skin_color: "fair",
    species: [
      "https://swapi.co/api/species/1/"
    ],
    starships: [
      "https://swapi.co/api/starships/12/",
      "https://swapi.co/api/starships/22/"
    ],
    url: "https://swapi.co/api/people/1/",
    vehicles: [
      "https://swapi.co/api/vehicles/14/",
      "https://swapi.co/api/vehicles/30/"
    ]
  }
];

export const mockMovieFetchResponse = {
  title: 'A New Hope',
  episode_id: 4,
  opening_crawl: 'It is a period of civil war.\r\nRebel spaceships, striking\r\nfrom a hidden base, have won\r\ntheir first victory against\r\nthe evil Galactic Empire.\r\n\r\nDuring the battle, Rebel\r\nspies managed to steal secret\r\nplans to the Empire\'s\r\nultimate weapon, the DEATH\r\nSTAR, an armored space\r\nstation with enough power\r\nto destroy an entire planet.\r\n\r\nPursued by the Empire\'s\r\nsinister agents, Princess\r\nLeia races home aboard her\r\nstarship, custodian of the\r\nstolen plans that can save her\r\npeople and restore\r\nfreedom to the galaxy....',
  director: 'George Lucas',
  producer: 'Gary Kurtz, Rick McCallum',
  release_date: '1977-05-25',
  characters:[
    'https://swapi.co/api/people/1/',
    'https://swapi.co/api/people/2/',
    'https://swapi.co/api/people/3/',
    'https://swapi.co/api/people/4/',
    'https://swapi.co/api/people/5/',
    'https://swapi.co/api/people/6/',
    'https://swapi.co/api/people/7/',
    'https://swapi.co/api/people/8/',
    'https://swapi.co/api/people/9/',
    'https://swapi.co/api/people/10/',
    'https://swapi.co/api/people/12/',
    'https://swapi.co/api/people/13/',
    'https://swapi.co/api/people/14/',
    'https://swapi.co/api/people/15/',
    'https://swapi.co/api/people/16/',
    'https://swapi.co/api/people/18/',
    'https://swapi.co/api/people/19/',
    'https://swapi.co/api/people/81/'
  ],
  planets:[
    'https://swapi.co/api/planets/2/',
    'https://swapi.co/api/planets/3/',
    'https://swapi.co/api/planets/1/'
  ],
  starships:[
    'https://swapi.co/api/starships/2/',
    'https://swapi.co/api/starships/3/',
    'https://swapi.co/api/starships/5/',
    'https://swapi.co/api/starships/9/',
    'https://swapi.co/api/starships/10/',
    'https://swapi.co/api/starships/11/',
    'https://swapi.co/api/starships/12/',
    'https://swapi.co/api/starships/13/'
  ],
  vehicles:[
    'https://swapi.co/api/vehicles/4/',
    'https://swapi.co/api/vehicles/6/',
    'https://swapi.co/api/vehicles/7/',
    'https://swapi.co/api/vehicles/8/'
  ],
  species:[
    'https://swapi.co/api/species/5/',
    'https://swapi.co/api/species/3/',
    'https://swapi.co/api/species/2/',
    'https://swapi.co/api/species/1/',
    'https://swapi.co/api/species/4/'
  ],
  created: '2014-12-10T14:23:31.880000Z',
  edited: '2015-04-11T09:46:52.774897Z',
  url: 'https://swapi.co/api/films/1/'
};

export const mockPlanetsResponse = {
  count: 61,
  next: 'https://swapi.co/api/planets/?page=2',
  previous: null,
  results: [
    {
      name: 'Alderaan',
      rotation_period: '24',
      orbital_period: '364',
      diameter: '12500',
      climate: 'temperate',
      gravity: '1 standard',
      terrain: 'grasslands, mountains',
      surface_water: '40',
      population: '2000000000',
      residents: [
        'https://swapi.co/api/people/5/',
        'https://swapi.co/api/people/68/',
        'https://swapi.co/api/people/81/'
      ],
      films: ['https://swapi.co/api/films/6/', 'https://swapi.co/api/films/1/'],
      created: '2014-12-10T11:35:48.479000Z',
      edited: '2014-12-20T20:58:18.420000Z',
      url: 'https://swapi.co/api/planets/2/'
    }
  ]
};

export const mockResidentResponse = {
  name: 'Leia Organa',
  height: '150',
  mass: '49',
  hair_color: 'brown',
  skin_color: 'light',
  eye_color: 'brown',
  birth_year: '19BBY',
  gender: 'female',
  homeworld: 'https://swapi.co/api/planets/2/',
  films:[
    'https://swapi.co/api/films/2/',
    'https://swapi.co/api/films/6/',
    'https://swapi.co/api/films/3/',
    'https://swapi.co/api/films/1/',
    'https://swapi.co/api/films/7/'
  ],
  species: ['https://swapi.co/api/species/1/'],
  vehicles: ['https://swapi.co/api/vehicles/30/'],
  starships: [],
  created: '2014-12-10T15:20:09.791000Z',
  edited: '2014-12-20T21:17:50.315000Z',
  url: 'https://swapi.co/api/people/5/'
};