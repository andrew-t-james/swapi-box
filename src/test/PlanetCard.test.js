import React from 'react';
import { shallow } from 'enzyme';
import PlanetCard from '../components/PlanetCard/PlanetCard';
import { mockCleanedPlanet } from '../mock-data/cleaned-data';
describe('PlanetCard', () => {
  let wrapper;
  const mockFunc = jest.fn();
  const mockProps = {
    name: 'Alderaan',
    population: '2000000000',
    climate: 'temperate',
    terrain: 'grasslands, mountains',
    residents: [undefined, undefined, undefined],
    favorite: false
  };

  beforeEach(() => wrapper = shallow(
    <PlanetCard
      {...mockProps}
      updateFavorites={mockFunc}
    />
  ));

  afterEach(() => wrapper.unmount());
  test('should call updateFavorites onClick', () => {
    wrapper.find('.card__favorite').first().simulate('click');
    expect(mockFunc).toHaveBeenCalled();
  });

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});