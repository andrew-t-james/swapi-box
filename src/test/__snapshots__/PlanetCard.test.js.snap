// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlanetCard should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlanetCard
    climate="temperate"
    favorite={false}
    name="Alderaan"
    population="2000000000"
    residents={
        Array [
            undefined,
            undefined,
            undefined,
          ]
    }
    terrain="grasslands, mountains"
    updateFavorites={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Alderaan
</h2>,
        <p>
          temperate
</p>,
        <p>
          2000000000
</p>,
        <p>
          
</p>,
        <p>
          grasslands, mountains
</p>,
        <span
          className="card__favorite"
          onClick={[Function]}
>
          <Button
                    name="⭐"
          />
</span>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Alderaan",
        },
        "ref": null,
        "rendered": "Alderaan",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "temperate",
        },
        "ref": null,
        "rendered": "temperate",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "2000000000",
        },
        "ref": null,
        "rendered": "2000000000",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            undefined,
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          undefined,
          undefined,
          undefined,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "grasslands, mountains",
        },
        "ref": null,
        "rendered": "grasslands, mountains",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Button
            name="⭐"
/>,
          "className": "card__favorite",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": "⭐",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Alderaan
</h2>,
          <p>
            temperate
</p>,
          <p>
            2000000000
</p>,
          <p>
            
</p>,
          <p>
            grasslands, mountains
</p>,
          <span
            className="card__favorite"
            onClick={[Function]}
>
            <Button
                        name="⭐"
            />
</span>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Alderaan",
          },
          "ref": null,
          "rendered": "Alderaan",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "temperate",
          },
          "ref": null,
          "rendered": "temperate",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "2000000000",
          },
          "ref": null,
          "rendered": "2000000000",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              undefined,
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            undefined,
            undefined,
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "grasslands, mountains",
          },
          "ref": null,
          "rendered": "grasslands, mountains",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Button
              name="⭐"
/>,
            "className": "card__favorite",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "name": "⭐",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
